api/api/src/test/scala/uk/ac/wellcome/platform/api/elasticsearch/WorksQueryTest.scala

```diff
@@ -153,6 +155,50 @@ class WorksQueryTest
       }
     }
 
+    it("searches the images.canonicalId as keyword") {
+      withLocalWorksIndex { index =>
+        val work = createIdentifiedWorkWith(
+          canonicalId = "abc123",
+          images = List(
+            createUnmergedImage.toIdentifiedWith(id = "def")
+          ))
+        val workNotMatching = createIdentifiedWorkWith(
+          canonicalId = "123abc",
+          images = List(
+            createUnmergedImage.toIdentifiedWith(id = "def456")
+          )
+        )
+        val query = "def"
+
+        insertIntoElasticsearch(index, work, workNotMatching)
+
+        assertResultsMatchForAllowedQueryTypes(index, query, List(work))
+      }
+    }
+
+    it("searches the images.sourceIdentifiers") {
+      withLocalWorksIndex { index =>
+        val work = createIdentifiedWorkWith(
+          canonicalId = "abc123",
+          images = List(
+            createUnmergedImageWith(identifierValue = "sourceIdentifier123").toIdentified
+          )
+        )
+        val workNotMatching = createIdentifiedWorkWith(
+          canonicalId = "123abc",
+          images = List(
+            createUnmergedImageWith(identifierValue = "sourceIdentifier456").toIdentified
+          )
+        )
+
+        val query = "sourceIdentifier123"
+
+        insertIntoElasticsearch(index, work, workNotMatching)
+
+        assertResultsMatchForAllowedQueryTypes(index, query, List(work))
+      }
+    }
+
```

Comment:
No test for `images.otherIdentifiers` because there is no way in which it can become populated at the moment - I left it in the query for consistency with other fields / in case it does become populated